### PR TITLE
fix: handle duplicate forkids

### LINKED ISSUE #5935: Using `vm.selectFork()` to change fork causes an application panic
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (96ab913 2023-09-28T00:16:57.440609799Z)

### What command(s) is the bug in?

forge test

### Operating System

Linux

### Describe the bug

I'm unable to switch forks created in forge when using `vm.selectFork(forkId)`. I get the following error:
```The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: crates/evm/src/executor/backend/mod.rs:663

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

   0: evm::cheatcodes::fork
      at crates/evm/src/executor/inspector/cheatcodes/fork.rs:26
   1: evm::cheatcodes::apply
      at crates/evm/src/executor/inspector/cheatcodes/mod.rs:213

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Aborted
```

Steps to reproduce:
1. Do `forge init` and add the following text into `test/TestFork.t.sol`:
```// SPDX-License-Identifier: NONE
pragma solidity >=0.8.0 <0.9.0;

import "forge-std/Test.sol";

contract SimpleStorage {
    uint256 public value;
    function set(uint256 _value) external {
        value = _value;
    }
}

/// @dev start anvil --port 35353
contract TestFork is Test {
    function testFork() public {
        uint256 forkId1 = vm.createFork("http://localhost:35353");
        uint256 forkId2 = vm.createFork("http://localhost:35353");
        vm.selectFork(forkId1);
        SimpleStorage myContract = new SimpleStorage();
        emit log_address(address(myContract));
        vm.selectFork(forkId2);
        SimpleStorage myContract2 = new SimpleStorage();
        emit log_address(address(myContract2));
    }
}
```
3. Start a local anvil node using `anvil --port 35353`
4. In another terminal, run `forge test --mc TestFork`

### PR DESCRIPTION
**Pull request by mattsse**
Closes #5935

we have an edge case with forkids internally that can lead to an out of bounds bug if the same URL is used twice and the local mapping is updated

https://github.com/foundry-rs/foundry/blob/master/crates/evm/src/executor/backend/mod.rs#L1624-L1634

### CODE REVIEW

**Comment by frank-lim-partior** [comment]
The expected behavior is that `forkId1 != forkId2`, is that correct? I had hoped to use this to test contracts across 2 different forks from the same state.

**Comment by mattsse** [comment]
this is rather a bug how me map the assigned uint256 forkids internally, because both forking off the same endpoint at the same height, so we need to change how we keep track of those, will fix shortly

**Comment by DaniPopes** [comment]
@mattsse ping

**Comment by mattsse** [comment]
@DaniPopes whoops, totally forgot about this.

I took a page out of the cargo playbook and used an URL hack to make them unique, this was easier than rewriting the internal mappings...
This is not ideal but at least fixes the issue, the fact that we only got one report so far indicates that this is not a common pattern so the fix should be fine

File: crates/forge/tests/it/repros.rs [hunk 1]
```diff
@@ -198,3 +198,6 @@ test_repro!(6180);
 
 // https://github.com/foundry-rs/foundry/issues/6437
 test_repro!(6437);
+
+// <https://github.com/foundry-rs/foundry/issues/5935>
+test_repro!(5935);
```
Comment (DaniPopes) [inline crates/forge/tests/it/repros.rs:203]: order :)

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by DaniPopes — APPROVED** [review]
